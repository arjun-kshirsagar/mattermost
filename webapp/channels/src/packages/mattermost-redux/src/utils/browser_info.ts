// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export function getBrowserInfo() {
    const userAgent = window.navigator.userAgent;

    let browser = 'Unknown';
    let browserVersion = 'Unknown';

    // Simple browser detection
    if (userAgent.includes('Firefox')) {
        browser = 'Firefox';
    } else if (userAgent.includes('Chrome')) {
        browser = 'Chrome';
    } else if (userAgent.includes('Safari')) {
        browser = 'Safari';
    } else if (userAgent.includes('Edge')) {
        browser = 'Edge';
    }

    // Get browser version
    const match = userAgent.match(/(firefox|chrome|safari|edge(?=\/))\/?\s*(\d+)/i);
    if (match) {
        browserVersion = match[2];
    }

    return {browser, browserVersion};
}

export function getPlatformInfo() {
    // Casting to undefined in case it is deprecated in any browser
    const platform = window.navigator.platform as string | undefined;

    let platformName = 'Unknown';
    if (platform?.toLowerCase().includes('win')) {
        platformName = 'Windows';
    } else if (platform?.toLowerCase().includes('mac')) {
        platformName = 'MacOS';
    } else if (platform?.toLowerCase().includes('linux')) {
        platformName = 'Linux';
    }

    return platformName;
}
