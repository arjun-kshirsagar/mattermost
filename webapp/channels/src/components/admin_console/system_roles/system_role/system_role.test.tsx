// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';

import {TestHelper} from 'utils/test_helper';

import SystemRole from './system_role';

// Mock the dependencies to avoid rendering issues
jest.mock('./system_role_permissions', () => 
    function MockSystemRolePermissions() {
        return <div data-testid="system-role-permissions" />;
    }
);

jest.mock('./system_role_users', () => 
    function MockSystemRoleUsers() {
        return <div data-testid="system-role-users" />;
    }
);

jest.mock('components/admin_console/save_changes_panel', () => 
    function MockSaveChangesPanel() {
        return <div data-testid="save-changes-panel" />;
    }
);

jest.mock('components/admin_console/blockable_link', () => (props: any) => (
    <a {...props} data-testid="blockable-link" />
));

describe('admin_console/system_role', () => {
    const props = {
        role: TestHelper.getRoleMock(),
        isDisabled: false,
        isLicensedForCloud: false,
        actions: {
            editRole: jest.fn(),
            updateUserRoles: jest.fn(),
            setNavigationBlocked: jest.fn(),
        },
    };

    const renderComponent = (customProps = {}) => {
        const mergedProps = {...props, ...customProps};
        return render(
            <MemoryRouter>
                <SystemRole {...mergedProps} />
            </MemoryRouter>
        );
    };

    test('should render correctly with default props', () => {
        const {container, getByTestId} = renderComponent();
        
        // Verify that the component renders properly
        expect(getByTestId('system-role-permissions')).toBeInTheDocument();
        expect(getByTestId('system-role-users')).toBeInTheDocument();
        expect(getByTestId('save-changes-panel')).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    test('should render correctly with isLicensedForCloud = true', () => {
        const {container, getByTestId} = renderComponent({isLicensedForCloud: true});
        
        // Verify that the component renders properly
        expect(getByTestId('system-role-permissions')).toBeInTheDocument();
        expect(getByTestId('system-role-users')).toBeInTheDocument();
        expect(getByTestId('save-changes-panel')).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
});
