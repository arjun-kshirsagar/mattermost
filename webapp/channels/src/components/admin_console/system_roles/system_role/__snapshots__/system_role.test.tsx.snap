// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin_console/system_role should render correctly with default props 1`] = `
<div>
  <div
    class="wrapper--fixed"
  >
    <div
      class="admin-console__header with-back"
    >
      <div>
        <a
          class="fa fa-angle-left back"
          data-testid="blockable-link"
          to="/admin_console/user_management/system_roles"
        />
        role_name
      </div>
    </div>
    <div
      class="admin-console__wrapper"
    >
      <div
        class="admin-console__content"
      >
        <div
          data-testid="system-role-permissions"
        />
        <div
          data-testid="system-role-users"
        />
      </div>
    </div>
    <div
      data-testid="save-changes-panel"
    />
  </div>
</div>
`;

exports[`admin_console/system_role should render correctly with isLicensedForCloud = true 1`] = `
<div>
  <div
    class="wrapper--fixed"
  >
    <div
      class="admin-console__header with-back"
    >
      <div>
        <a
          class="fa fa-angle-left back"
          data-testid="blockable-link"
          to="/admin_console/user_management/system_roles"
        />
        role_name
      </div>
    </div>
    <div
      class="admin-console__wrapper"
    >
      <div
        class="admin-console__content"
      >
        <div
          data-testid="system-role-permissions"
        />
        <div
          data-testid="system-role-users"
        />
      </div>
    </div>
    <div
      data-testid="save-changes-panel"
    />
  </div>
</div>
`;
